game/src/layer/map.rs

```diff
@@ -250,17 +250,18 @@ impl Static {
         )
     }
 
+    // TODO update the colors to cover more cases
```

Comment:
This actually crashes on the montlake Seattle map now, because one of the categories doesn't match up to a color, and `ColorDiscrete` blindly assumes there's a color defined for every string category you pass in. We can just rewrite this entire function:

```
    pub fn amenities(ctx: &mut EventCtx, app: &App) -> Static {
        let food = Color::RED;
        let school = Color::CYAN;
        let shopping = Color::PURPLE;
        let other = Color::GREEN;

        let mut unzoomed = GeomBatch::new();
        let mut zoomed = GeomBatch::new();
        for b in app.primary.map.all_buildings() {
            if b.amenities.is_empty() {
                continue;
            }
            let mut color = None;
            for a in &b.amenities {
                if let Some(t) = AmenityType::categorize(&a.amenity_type) {
                    color = Some(match t {
                        AmenityType::Food => food,
                        AmenityType::School => school,
                        AmenityType::Shopping => shopping,
                        _ => other,
                    });
                    break;
                }
            }
            let color = color.unwrap_or(other);
            unzoomed.push(color, b.polygon.clone());
            zoomed.push(color.alpha(0.4), b.polygon.clone());
        }

        let panel = Panel::new(Widget::col(vec![
            header(ctx, "Amenities"),
            ColorLegend::row(ctx, food, AmenityType::Food.to_string()),
            ColorLegend::row(ctx, school, AmenityType::School.to_string()),
            ColorLegend::row(ctx, shopping, AmenityType::Shopping.to_string()),
            ColorLegend::row(ctx, other, "other".to_string()),
        ]))
        .aligned_pair(PANEL_PLACEMENT)
        .build(ctx);

        Static {
            panel,
            unzoomed: ctx.upload(unzoomed),
            zoomed: ctx.upload(zoomed),
            name: "amenities",
        }
    }
```
(I really need to learn how to push a commit onto somebody else's PR...)

This rewrite just assigns a color to a few "important"?? categories, and leaves all others as a generic color. This layer can be accessed by running A/B Street (`game` crate), opening layers (hotkey `l`), then amenities (hotkey `a`). I don't know if many people use this layer often; I occasionally open it to quickly understand where businesses are clustered. Picking a different color for every category isn't a useful or maintainable exercise, and this layer never attempted to solve the problem of one building hosting multiple types of shop. If you have better ideas what to do with it, we could try them later, but I don't think this layer is important at all.

Reply:
Thank you for the rewrite and the explantation, I had not played with that layer yet, it makes sense now.